test/jdk/javax/imageio/plugins/wbmp/WBMPStreamTruncateTest.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug     8266435
+ * @summary Test verifies that WBMPImageReader doesnt truncate
+ *          the stream and reads it fully
+ * @run     main WBMPStreamTruncateTest
+ */
+
+import java.awt.Color;
+import java.awt.Graphics2D;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import javax.imageio.ImageIO;
+import javax.imageio.stream.ImageInputStreamImpl;
+
+public class WBMPStreamTruncateTest
+{
+    static final int LIMIT = 100;
+    static final int width = 100;
+    static final int height = 100;
+    public static void main(String[] args) throws IOException
+    {
+        String sep = System.getProperty("file.separator");
+        String dir = System.getProperty("test.src", ".");
+        String filePath = dir+sep;
+        BufferedImage srcImage = new
+                BufferedImage(width, height, BufferedImage.TYPE_BYTE_BINARY);
+        Graphics2D g = (Graphics2D) srcImage.getGraphics();
+        g.setBackground(Color.WHITE);
+        g.fillRect(0, 0, srcImage.getWidth(), srcImage.getHeight());
+        g.dispose();
+        // create WBMP image
+        File imageFile = File.
+                createTempFile("test", ".wbmp", new File(filePath));
+        imageFile.deleteOnExit();
```

Comment:
In one another review I saw the deleteOnExit() usage, I remember that in case of full testrun via makefile, such files were not deleted, can you please confirm that it is work fine? probably it that was fixed already. Note that the full test run uses custom tmp folder inside the result dir.

Reply:
@mrserb I ran the test using "make test" and i dont see any .wbmp file after test run in results directory.
Also i verified that it doesn't leave any image in test directory of CI machines.

Reply:
Ok, thank you for confirmation.